processing/src/main/java/io/druid/segment/serde/DoubleGenericColumnPartSerdeV2.java

```diff
@@ -0,0 +1,188 @@
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.serde;
+
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Suppliers;
+import com.google.common.primitives.Ints;
+import io.druid.collections.bitmap.ImmutableBitmap;
+import io.druid.java.util.common.IAE;
+import io.druid.java.util.common.io.smoosh.FileSmoosher;
+import io.druid.segment.DoubleColumnSerializer;
+import io.druid.segment.column.ValueType;
+import io.druid.segment.data.BitmapSerde;
+import io.druid.segment.data.BitmapSerdeFactory;
+import io.druid.segment.data.ByteBufferWriter;
+import io.druid.segment.data.CompressedDoublesIndexedSupplier;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.channels.WritableByteChannel;
+
+public class DoubleGenericColumnPartSerdeV2 implements ColumnPartSerde
+{
+  static final byte VERSION_ONE = 0x1;
+  private final ByteOrder byteOrder;
+  private Serializer serialize;
+  private final BitmapSerdeFactory bitmapSerdeFactory;
+
+  @JsonCreator
+  public static DoubleGenericColumnPartSerdeV2 getDoubleGenericColumnPartSerde(
+      @JsonProperty("byteOrder") ByteOrder byteOrder,
+      @Nullable @JsonProperty("bitmapSerdeFactory") BitmapSerdeFactory bitmapSerdeFactory
+  )
+  {
+    return new DoubleGenericColumnPartSerdeV2(byteOrder,
+                                              bitmapSerdeFactory != null
+                                              ? bitmapSerdeFactory
+                                              : new BitmapSerde.LegacyBitmapSerdeFactory(), null
+    );
+  }
+
+  @JsonProperty
+  public ByteOrder getByteOrder()
+  {
+    return byteOrder;
+  }
+
+
+  public DoubleGenericColumnPartSerdeV2(
+      ByteOrder byteOrder,
+      BitmapSerdeFactory bitmapSerdeFactory,
+      Serializer serialize
+  )
+  {
+    this.byteOrder = byteOrder;
+    this.bitmapSerdeFactory = bitmapSerdeFactory;
+    this.serialize = serialize;
+  }
+
+  @Override
+  public Serializer getSerializer()
+  {
+    return serialize;
+  }
+
+  @Override
+  public Deserializer getDeserializer()
+  {
+    return (buffer, builder, columnConfig) -> {
+      byte versionFromBuffer = buffer.get();
+
+      if (VERSION_ONE == versionFromBuffer) {
```

Comment:
why/how will have a version missmatch? the Serdeser is done by jackson anyway that has the mapping between factories and type name.